catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -147,6 +148,33 @@ public void init() {
         });
   }
 
+  @Override
+  public Object apply(Object req) {
+    if (!(req instanceof List)) {
+      throw new IllegalArgumentException("params not list");
+    }
+
+    List params = (List) req;
+
+    if (params.size() != 1) {
+      throw new IllegalArgumentException("must pass exactly 1 param");
+    }
+
+    Object param = params.get(0);
+
+    if (!(param instanceof String)) {
+      throw new IllegalArgumentException("param not string");
+    }
+
+    try {
+      CqlRequest cqlRequest = mapper.readValue((String) param, CqlRequest.class);
+      return executeCqlQuery(cqlRequest);
+    } catch (Exception e) {
```

Comment:
Can we avoid catching a generic exception here?

Reply:
Looks like we could do
`(UnsupportedQueryException | SourceUnavailableException | FederationException e)`

Reply:
Barring any other potential runtime exceptions